# HG changeset patch
# User Ben Abbott <bpabbott@mac.com>
# Date 1241831465 14400
#      Fri May 08 21:11:05 2009 -0400
# Node ID ad33527d2e51021354ec3a5a76b1f4592e7792cb
# Parent  7a10410db2c68858f81bfdc7f94df6e57c25744b
Have 'legend' inherit font properties from the parent axis.

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,13 @@
+2009-05-08  Ben Abbott <bpabbott@mac.com>
+
+	* plot/__gnuplot_has_feature__.m: Add feature "key_has_font_properties".
+	* plot/__go_draw_axes__.m: Have legend inherit the axis font-name/size.
+
 2009-05-04  Peter O'Gorman  <pogma@thewrittenword.com>
 
 	* miscellaneous/tar.m, miscellaneous/unpack.m:
 	Pass "xvf" to tar instead of "-x -v -f".
 
 2009-04-30  Ben Abbott <bpabbott@mac.com>
 
 	* plot/__gnuplot_has_feature__.m: Change version for 
diff --git a/scripts/plot/__gnuplot_has_feature__.m b/scripts/plot/__gnuplot_has_feature__.m
--- a/scripts/plot/__gnuplot_has_feature__.m
+++ b/scripts/plot/__gnuplot_has_feature__.m
@@ -25,22 +25,23 @@
 function res = __gnuplot_has_feature__ (feature)
   persistent features has_features
   features = {"x11_figure_position",
               "transparent_patches",
               "transparent_surface",
               "epslatex_implies_eps_filesuffix",
               "epslatexstandalone_terminal",
               "screen_coordinates_for_{lrtb}margin",
-              "variable_GPVAL_TERMINALS"};
+              "variable_GPVAL_TERMINALS",
+              "key_has_font_properties"};
 
   if (isempty (has_features))
     gnuplot_version = __gnuplot_version__ ();
-    versions = {"4.3", "4.3", "4.3", "4.2", "4.2", "4.3", "4.3"};
-    operators = {">=", ">=", ">=", ">=", ">=", ">=", ">="};
+    versions = {"4.3", "4.3", "4.3", "4.2", "4.2", "4.3", "4.3", "4.3"};
+    operators = {">=", ">=", ">=", ">=", ">=", ">=", ">=", ">="};
     have_features = logical (zeros (size (features)));
     for n = 1 : numel (have_features)
       has_features(n) = compare_versions (gnuplot_version, versions{n}, operators{n});
     endfor
   endif
 
   n = find (strcmpi (feature, features));
   if (isempty (n))
diff --git a/scripts/plot/__go_draw_axes__.m b/scripts/plot/__go_draw_axes__.m
--- a/scripts/plot/__go_draw_axes__.m
+++ b/scripts/plot/__go_draw_axes__.m
@@ -1204,17 +1204,23 @@ function __go_draw_axes__ (h, plot_strea
 	case "best" 
 	  pos = "";
 	  warning ("legend: 'Best' not yet implemented for location specifier.\n");
 	  ## Least conflict with data in plot.
 	  ## Least unused space outside plot.
 	otherwise
 	  pos = "";
       endswitch
-      fprintf (plot_stream, "set key %s %s %s %s;\n", inout, pos, box, reverse);
+      if (__gnuplot_has_feature__ ("key_has_font_properties"))
+        fontspec = create_fontspec (axis_obj.fontname, axis_obj.fontsize);
+      else
+	fontspec = "";
+      endif
+      fprintf (plot_stream, "set key %s %s %s %s %s;\n", inout, pos, box,
+               reverse, fontspec);
     else
       fputs (plot_stream, "unset key;\n");
     endif
 
     fputs (plot_stream, "set style data lines;\n");
 
     if (! use_gnuplot_for_images)
       for i = 1:ximg_data_idx
